src/attributeformmodelbase.cpp

```diff
@@ -214,16 +214,15 @@ void AttributeFormModelBase::updateAttributeValue( QStandardItem* item )
   }
 }
 
-void AttributeFormModelBase::flatten( QgsAttributeEditorContainer* container, QStandardItem* parent, const QString& visibilityExpressions, QVector<QStandardItem*>& items )
+void AttributeFormModelBase::flatten( QgsAttributeEditorContainer* container, QStandardItem* parent, const QString& parentVisibilityExpressions, QVector<QStandardItem*>& items )
 {
-  QString visibilityExpression = visibilityExpressions;
-
   Q_FOREACH( QgsAttributeEditorElement* element, container->children() )
   {
     switch ( element->type() )
     {
       case QgsAttributeEditorElement::AeTypeContainer:
       {
+        QString visibilityExpression = parentVisibilityExpressions;
         QgsAttributeEditorContainer* container = static_cast<QgsAttributeEditorContainer*>( element );
         if ( container->visibilityExpression().enabled() )
         {
```

Comment:
where is `visibilityExpression` initialized then? (line 229)

Reply:
With the `parentVisibilityExpression` - and this is initialized on [call of the function](https://github.com/opengisch/QField/pull/205/files/de6c6738911ce289f44ec899cb5cb16c10e2e4ce#diff-e41b3ba57e866b91157e3b015593a8caL158)
Can't we do that?